### PR TITLE
rgw/rgw_rados: do not omap_getvals with (u64)-1 max (updated)

### PR DESCRIPTION
**Pull request by yehudasa**
Fixes: http://tracker.ceph.com/issues/17985

### CODE REVIEW

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -11296,18 +11296,21 @@ int RGWRados::omap_get_all(rgw_obj& obj, bufferlist& header,
...
     std::map<string, bufferlist> t;
-    r = ref.ioctx.omap_get_vals(ref.oid, start_after, count, &m);
+    r = ref.ioctx.omap_get_vals(ref.oid, start_after, count, &t);
     if (r < 0) {
       return r;
     }
     if (t.empty()) {
       break;
     }
+    done = (t.size() < count);
```
Comment (liewegas) [inline src/rgw/rgw_rados.cc]: It's possible for the OSD to return fewer entries and not be at the end.

Comment (liewegas) [inline src/rgw/rgw_rados.cc]: (For example, there is a now an omap get limit that's enforced at the osd to work around client code like this)

Comment (yehudasa) [inline src/rgw/rgw_rados.cc]: @liewegas in that case, we need to have omap_getvals() to also return is_truncated

**Comment by yehudasa** [comment]
@liewegas updated

**Comment by dachary** [comment]
jenkins test this please (https://jenkins.ceph.com/job/ceph-pull-requests/15081/console fails readable.sh)

**Comment by dachary** [comment]
jenkins test this please (osd-crush.sh)

**Comment by mattbenjamin** [comment]
passed my teuthology run

**Pull request merged**